packages/unified-latex-util-arguments/libs/gobble-single-argument.ts

```diff
@@ -212,3 +220,69 @@ function partialStringMatch(node: Ast.Node, token: Ast.Node): boolean {
         node.content.length > token.content.length
     );
 }
+
+function cloneStringNode(node: Ast.String, content: string): Ast.String {
+    return Object.assign({}, node, { content });
+}
+
+// Find the position of the open brace and the closing brace.
+// Returns undefined if the brace isn't found.
+// This may mutate `nodes`.
+function findBracePositions(nodes: Ast.Node[], startPos: number,
+    openMark: string, closeMark?: string): [number, number] | undefined {
+    const currNode = nodes[startPos];
+    if (!openMark || !match.anyString(currNode)) return;
+    const nodeContent = currNode.content;
+    // The first node we encounter must contain the opening brace.
+    if (!nodeContent.startsWith(openMark)) return;
+    const openMarkPos = startPos;
+    if (currNode.content.length > openMark.length) {
+        const nodeContent = currNode.content;
+        currNode.content = openMark;
+        nodes.splice(openMarkPos + 1, 0,
+            cloneStringNode(currNode, nodeContent.slice(openMark.length))
+        );
+    }
+    if (!closeMark) {
+        // In such a case, the token immediately preceding the opening brace
+        // will be treated as an argument. If the next token is a string node,
+        // only its first character is picked up.
+        let closeMarkPos = openMarkPos + 1;
+        const argNode = nodes[closeMarkPos];
+        if (!argNode) return;
+        if (match.anyString(argNode) && argNode.content.length > 1) {
+            const argContent = argNode.content;
+            argNode.content = argContent[0];
+            nodes.splice(closeMarkPos + 1, 0, cloneStringNode(argNode, argContent.slice(1)));
+        }
+        return [openMarkPos, closeMarkPos];
+    }
+    for (let i = openMarkPos + 1; i < nodes.length; i++) {
+        const node = nodes[i];
+        if (!match.anyString(node)) continue;
+        const nodeContent = node.content;
+        let closeMarkIndex = nodeContent.indexOf(closeMark);
+        if (closeMarkIndex === -1) continue;
+        // Found a closeMark, store its index to `closeMarkPos`
+        let closeMarkPos = i;
+        if (nodeContent.length > closeMark.length) {
+            node.content = closeMark;
+            let prev = nodeContent.slice(0, closeMarkIndex);
+            let next = nodeContent.slice(closeMarkIndex + closeMark.length);
+            if (prev) {
+                nodes.splice(closeMarkPos, 0, cloneStringNode(node, prev));
+                closeMarkPos++;
+            }
+            if (next) {
+                nodes.splice(closeMarkPos + 1, 0, cloneStringNode(node, next));
+            }
+        }
+        return [openMarkPos, closeMarkPos];
+    }
+}
+function stringifyGroup(group: ArgSpec.Group): string {
```

Comment:
I think the `printRaw` function already does this.

Reply:
Thank you for point out. I believe its fixed now.